File: src/main/programs/Java/NearestPalindrome.java
package Java;

public class NearestPalindrome {
    public static boolean isPalindrome(int n) {
        String s = String.valueOf(n);
        return s.equals(new StringBuilder(s).reverse().toString());
    }

    public static int nearestPalindrome(int n) {
        int smaller = n - 1, larger = n + 1;
        while (true) {
            if (isPalindrome(smaller)) return smaller;
            if (isPalindrome(larger)) return larger;
            smaller--;
            larger++;
        }
    }

    public static void main(String[] args) {
        int num = 123;
        System.out.println("Nearest Palindrome: " + nearestPalindrome(num));
    }
}

File: src/main/programs/NearestPalindrome.kt
fun isPalindrome(n: Int): Boolean {
    val s = n.toString()
    return s == s.reversed()
}

fun nearestPalindrome(n: Int): Int {
    var smaller = n - 1
    var larger = n + 1

    while (true) {
        if (isPalindrome(smaller)) return smaller
        if (isPalindrome(larger)) return larger
        smaller--
        larger++
    }
}

fun main() {
    val num = 123
    println("Nearest Palindrome: ${nearestPalindrome(num)}")
}